cadquery/occ_impl/geom.py

```diff
@@ -143,9 +143,31 @@ def distanceToPlane(self):
         raise NotImplementedError(
             "Have not needed this yet, but FreeCAD supports it!")
 
-    def projectToPlane(self):
-        raise NotImplementedError(
-            "Have not needed this yet, but FreeCAD supports it!")
+    def projectToPlane(self, *args):
+        """
+        Vector is projected onto the plane provided as input.
+
+        :param args: Plane object or base and normal vectors that define the plane
+
+        This method modifies the vector in place and returns the new vector.
+        """
+        if len(args) == 2:
+            base = args[0]
+            normal = args[1]
+        elif len(args) == 1:
+            base = args[0].origin
+            normal = args[0].zDir
+        else:
+            raise TypeError("Expected 1 or 2 arguments consisting of 1 Plane object \
+or a base Vector and a normal Vector object.")
+
+        result = self-normal*(((self-base).dot(normal))/normal.Length**2)
```

Comment:
I'd return here: 
```python
return self-normal*(((self-base).dot(normal))/normal.Length**2)
```
so no in-place modification will surprise our users.

Reply:
Agreed.